students/17th/team3/binhosong/users/views.py

```diff
@@ -0,0 +1,73 @@
+import json
+import bcrypt
+import jwt
+import my_settings
+
+from django.http        import JsonResponse
+from django.views       import View
+from django.db.models   import Q
+
+from .models            import Account
+
+MINIMUM_PASSWORD_LENGTH = 8
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+
+        try : 
+            # 이메일에 '@' , '.' 여부 확인 
+            if '@' not in data['email'] or '.' not in data['email']:
+                return JsonResponse({'message':'INVALID_EMAIL'}, status=400)
+
+            # Password length Check
+            if len(data['password']) < MINIMUM_PASSWORD_LENGTH:
+                return JsonResponse({'message':'SHORT_PASSWORD'}, status=400)
+        
+            # Email 중복 확인 
+            if Account.objects.filter(email=data['email']).exists():
+                return JsonResponse({'message':'USER_ALREADY_EXISTS'}, status=409)
+
+            if Account.objects.filter(phone=data['phone']).exists():
+                return JsonResponse({'message':'PHONE_NUMBER_ALREADY_EXISTS'}, status=409)
+
+            if Account.objects.filter(nickname=data['nickname']).exists():
+                return JsonResponse({'message':'NICKNAME_ALREADY_EXISTS'}, status=409)
+
+            hash_pw = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+
+            Account.objects.create(
+                    email     = data['email'],
+                    name      = data['name'],
+                    nickname  = data['nickname'],
+                    password  = hash_pw,
+                    phone     = data['phone']
+            )
+            return JsonResponse({'message' : 'SUCCESS'}, status=201)
+
+        except KeyError:
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+
+class LoginView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        
+        try :
+            email    = data.get('email')
+            phone    = data.get('phone')
+            password = data.get('password')
+
+            if Account.objects.filter(Q(email=email) | Q(phone=phone)).exists():
+                account = Account.objects.get(Q(email=email) | Q(phone=phone))
+
+                if bcrypt.checkpw(password.encode('utf-8'), account.password.encode('utf-8')):
+                    token = jwt.encode({'email' : email}, my_settings.SECRET['secret'], algorithm = 'HS256')
+                    return JsonResponse({'message' : 'SUCCESS'}, status=200)
+                else:
+                    return JsonResponse({"message": "INVALID_PASSWORD"}, status=401)
+            else:
+                return JsonResponse({"message": "INVALID_USER"}, status=401)
```

Comment:
```suggestion      
                return JsonResponse({"message": "INVALID_PASSWORD"}, status=401)
          
            return JsonResponse({"message": "INVALID_USER"}, status=401)
```
줄일 수 있는 else:는 제거하고 인덴트로 해결하는게 좋습니다. 코드라인수를 줄일 수 있는 방법이기도 합니다.

Reply:
불필요한 else 제거하고 블로그 작성하였습니다 .
https://velog.io/@binogood68/Django-Backend-Westargram-%EA%B8%B0%EC%B4%88-%EC%84%A4%EC%A0%95
https://velog.io/@binogood68/Django-Backend-Westargram-2%EC%9E%A5-Mysql%EC%84%A4%EC%A0%95-%EB%B0%8F-%ED%9A%8C%EC%9B%90%EA%B0%80%EC%9E%85-%EA%B5%AC%ED%98%84
https://velog.io/@binogood68/Django-Backend-Westargram-3%EC%9E%A5%EB%A1%9C%EA%B7%B8%EC%9D%B8-%EA%B5%AC%ED%98%84